demos/python_demos/handwritten_japanese_recognition_demo/handwritten_japanese_recognition_demo.py

```diff
@@ -0,0 +1,116 @@
+"""
+
+from __future__ import print_function
+import os
+import sys
+import time
+import logging as log
+from argparse import ArgumentParser, SUPPRESS
+
+import cv2
+import numpy as np
+
+from openvino.inference_engine import IENetwork, IECore
+from utils.codec import CTCCodec
+
+
+def build_argparser():
+    parser = ArgumentParser(add_help=False)
+    args = parser.add_argument_group('Options')
+    args.add_argument('-h', '--help', action='help', default=SUPPRESS,
+                      help='Show this help message and exit.')
+    args.add_argument("-m", "--model", type=str, required=True,
+                      help="Path to an .xml file with a trained model.")
+    args.add_argument("-i", "--input", type=str, nargs="+", required=False,
+                      help="Required. Path to an image to infer")
+    args.add_argument('-ih', '--input-height', default=96, type=int,
+                      help='IR input image height')
+    args.add_argument('-iw', '--input-width', default=2000, type=int,
+                      help='IR input image width')
+    args.add_argument("-d", "--device", type=str, default="CPU",
+                      help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL, MYRIAD or HETERO: is "
+                           "acceptable. The sample will look for a suitable plugin for device specified. Default "
+                           "value is CPU")
+    args.add_argument("-ni", "--number_iter", type=int, default=1,
+                      help="Optional. Number of inference iterations")
+    args.add_argument("-cl", "--charlist", type=str, default="data/kondate_nakayosi_char_list.txt", help="Path to the decoding char list file")
+    return parser
+
+
+def get_characters(args):
+    '''Get characters'''
+    with open(args.charlist, 'r', encoding='utf-8') as f:
+        return ''.join(line.strip('\n') for line in f)
+
+
+def preprocess_input(image_name, height, width):
+    src = cv2.imread(image_name, cv2.IMREAD_GRAYSCALE)
+    ratio = float(src.shape[1]) / float(src.shape[0])
+    th = height
+    tw = int(th * ratio)
+    rsz = cv2.resize(src, (tw, th), interpolation=cv2.INTER_AREA).astype(np.float32)
+    # [h,w] -> [c,h,w]
+    img = rsz[None, :, :]
+    _, h, w = img.shape
+    # right edge padding
+    pad_img = np.pad(img, ((0, 0), (0, height - h), (0, width -  w)), mode='edge')
+    return pad_img
+
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+    model_xml = args.model
+    model_bin = os.path.splitext(model_xml)[0] + ".bin"
+
+    # Plugin initialization
+    ie = IECore()
+    # Read IR
+    log.info("Loading network files:\n\t{}\n\t{}".format(model_xml, model_bin))
+    net = IENetwork(model=model_xml, weights=model_bin)
+
+    assert len(net.inputs) == 1, "Demo supports only single input topologies"
+    assert len(net.outputs) == 1, "Demo supports only single output topologies"
```

Comment:
Hi. Thanks for the careful review. Assertion between charlist length and output shape is added. The one channel and one batch size has already been assured by line 92:
` input_image = preprocess_input(args.input[0], height=input_height, width=input_width)[None,:,:,:]
`   
Is the assertion necessary here?

Reply:
I'm not sure I understand how it assures that. `preprocess_input()` reads an image as gray scale, but if one provides a net that accepts BGR, the demo will try to put one channel image into an input with tree channels. So I suggest checking that number of channels for net's input is 1.

The similar thing with batch: if the model provided to IR has batch not 1, the demo will try to fill (batch,:,:,:) tensor with (1,:,:,:) tensor.

BTW, `nargs="+", ` should be removed from
```py
    args.add_argument("-i", "--input", type=str, nargs="+", required=True,
                      help="Required. Path to an image to infer")
```
because the demo expects a single image.